### PR TITLE
Updated interop tests with support for compression.

### PR DESCRIPTION
**Pull request by dgquintas**
Incluiding UNCOMPRESSABLE payload type, which relies on a 512KB file with data from /dev/urandom, rnd.dat, also included in this PR.

This PR includes #2586 (to be merged separately).

### CODE REVIEW

**Comment by yang-g** [comment]
Sorry for the delay. The code looks fine but the change breaks the spec in https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md

I would suggest adding new test cases to the spec instead of altering the existing test case since there are existing implementations in other languages also.

**Comment by dgquintas** [comment]
Done. PTAL. I'll update the documentation if everything looks fine.

**Comment by dgquintas** [comment]
PTAL after offline discussion about unnecessary changes to server.

File: src/core/surface/call.c [hunk 1]
```diff
@@ -532,6 +536,43 @@ grpc_compression_algorithm grpc_call_get_compression_algorithm(
...
+  /* Always support no compression */
+  GPR_BITSET(&call->encodings_accepted_by_peer, GRPC_COMPRESS_NONE);
+  for (i = 0; i < accept_encoding_parts.count; i++) {
+    const gpr_slice* slice = &accept_encoding_parts.slices[i];
+    if (grpc_compression_algorithm_parse(
+            (const char *)GPR_SLICE_START_PTR(*slice), GPR_SLICE_LENGTH(*slice),
+            &algorithm)) {
+      GPR_BITSET(&call->encodings_accepted_by_peer, algorithm);
+    } else {
+      /* TODO(dgq): it'd be nice to have a slice-to-cstr function to easily
```
Comment (yang-g) [inline src/core/surface/call.c]: gpr_dump_slice() in support/string.h?

File: test/cpp/interop/interop_client.cc [hunk 2]
```diff
@@ -103,9 +109,55 @@ void InteropClient::PerformLargeUnary(SimpleRequest* request,
   Status s = stub->UnaryCall(&context, *request, response);
 
   AssertOkOrPrintErrorStatus(s);
-  GPR_ASSERT(response->payload().type() == PayloadType::COMPRESSABLE);
-  GPR_ASSERT(response->payload().body() ==
-             grpc::string(kLargeResponseSize, '\0'));
+}
+
+// Shared code to set large payload, make rpc and check response payload.
+void InteropClient::PerformLargeCompressedUnary(SimpleRequest* request,
```
Comment (yang-g) [inline test/cpp/interop/interop_client.cc]: Remove this function and merge the checks with PerformLargeUnary?

Comment (dgquintas) [inline test/cpp/interop/interop_client.cc]: Done.

File: src/core/surface/call.c (hunk 1 above)
Comment (dgquintas) [inline src/core/surface/call.c]: Ah, nice! Done

File: test/cpp/interop/interop_client.cc [hunk 3]
```diff
@@ -95,6 +101,7 @@ void InteropClient::PerformLargeUnary(SimpleRequest* request,
   std::unique_ptr<TestService::Stub> stub(TestService::NewStub(channel_));
 
   ClientContext context;
+  InteropClientContextInspector inspector(context);
   request->set_response_type(PayloadType::COMPRESSABLE);
```
Comment (yang-g) [inline test/cpp/interop/interop_client.cc]: This response_type should be removed.

Comment (dgquintas) [inline test/cpp/interop/interop_client.cc]: It was present before

Comment (dgquintas) [inline test/cpp/interop/interop_client.cc]: Done. PTAL

**Comment by yang-g** [comment]
LGTM. Can we add a comment at the last line you removed saying if the response_type is not set, it defaults to COMPRESSABLE? Note in places we set it explicitly but not in DoLargeUnary().

**Comment by dgquintas** [comment]
Done.

**Comment by yang-g** [comment]
The sanity tests are failing, maybe try regenerating the project files?

**Comment by dgquintas** [comment]
Merged with master, regenerated (Grpc.mak was outdated) and rebuilt on Jenkins... let's see

**Comment by dgquintas** [comment]
Merged.

**Comment by dgquintas** [comment]
Ping :)

**Comment by yang-g** [comment]
It seems the sanity is failing. Could you take a look?

**Comment by dgquintas** [comment]
Yeah, I re-merged and re-run the tests. It's about to finish. So far so green. PTAL.

**Comment by dgquintas** [comment]
Jenkins is finally green-ish

**Pull request merged**